src/box/txn_limbo.c

```diff
@@ -324,9 +324,15 @@ txn_limbo_wait_complete(struct txn_limbo *limbo, struct txn_limbo_entry *entry)
 	assert(txn_has_flag(entry->txn, TXN_WAIT_SYNC));
 	double start_time = fiber_clock();
 	while (true) {
-		double deadline = start_time + replication_synchro_timeout;
-		double timeout = deadline - fiber_clock();
-		int rc = fiber_cond_wait_timeout(&limbo->wait_cond, timeout);
+		int rc;
+		if (replication_synchro_timeout_rollback_enabled) {
+			double timeout = start_time +
+				replication_synchro_timeout - fiber_clock();
+			rc = fiber_cond_wait_timeout(
+				&limbo->wait_cond, timeout);
+		} else {
+			rc = fiber_cond_wait(&limbo->wait_cond);
```

Comment:
Does this mean a synchro txn might be hanging for indefinite time until CONFIRM/PROMOTE comes? In my older comment I probably should have been clearer, sorry 😞 - I've meant that after the timeout we return control to the user and raise an error, but the txn remains in the limbo.

My concern with the current "infinite waiting" approach is that the only way to unblock the fiber now is to cancel it completely (if waiting never ends due to any reason). Even `box.begin{timeout = ... }` won't work because it affects only txn runtime, not its commit time.

OTOH, I am fine with leaving this as is. It is purely a user-experience thing. Thus up to @sergepetrenko.

Reply:
AFAIU, we wanted to make this a separate patch, we already have an issue for it https://github.com/tarantool/tarantool/issues/10181.

Reply:
Yes, this'll be a separate patch.
I think it's not a problem to do this separately, because the new behaviour is turned off by default. So the existing users won't have problems.